Source/driver/Castro_advance_ctu.cpp

```diff
@@ -157,11 +157,18 @@ Castro::retry_advance_ctu(Real dt, const advance_status& status)
 
     if (do_retry) {
 
-        if (status.suggested_dt > 0.0_rt && status.suggested_dt < dt) {
-            dt_subcycle = status.suggested_dt;
+        int max_level_to_advance = level;
+
+        if (parent->subcyclingMode() == "None" && level == 0) {
+            max_level_to_advance = parent->finestLevel();
         }
-        else {
-            dt_subcycle = std::min(dt, dt_subcycle) * retry_subcycle_factor;
+
+        for (int lev = level; lev <= max_level_to_advance; ++lev) {
+            if (status.suggested_dt > 0.0_rt && status.suggested_dt < dt) {
+                getLevel(lev).dt_subcycle = status.suggested_dt;
+            } else {
+                getLevel(lev).dt_subcycle = std::min(dt, dt_subcycle) * retry_subcycle_factor;
```

Comment:
This was setting `dt_subcycle` for the finer levels to be different from the value for level 0. It doesn't appear to fix the negative densities, though.
```suggestion
                getLevel(lev).dt_subcycle = std::min(dt, getLevel(lev).dt_subcycle) * retry_subcycle_factor;
```